File: app/src/main/java/com/gazilla/mihail/gazillaj/ui/main/contacts/ContactsFragment.java
package com.gazilla.mihail.gazillaj.ui.main.contacts;

import android.app.Activity;
import android.content.Intent;
import android.content.pm.ResolveInfo;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.gazilla.mihail.gazillaj.R;

import java.util.List;
/** фрагмент с контактами */
public class ContactsFragment extends Fragment {

    private static final String VK_APP_PACKAGE_ID = "com.vkontakte.android";
    private static final String FACEBOOK_APP_PACKAGE_ID = "com.facebook.katana";

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.contacts_fragment, null);

        /** Поля с соответсвующими ссылками */
        ConstraintLayout clOpenVK = view.findViewById(R.id.clOpenVK);
        ConstraintLayout clOpenFB = view.findViewById(R.id.clOpenFB);
        ConstraintLayout clOpenInsta = view.findViewById(R.id.clOpenInsta);
        ConstraintLayout clOpenMap = view.findViewById(R.id.clOpenMap);
        ConstraintLayout clOpenWWW = view.findViewById(R.id.clOpenWWW);
        ConstraintLayout clCallMe = view.findViewById(R.id.clCallMe);

        clOpenVK.setOnClickListener(v -> {
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://vk.com/gazilla_gz"));
            startActivity(intent);
        });

        clOpenFB.setOnClickListener(v -> {
            openLink(getActivity(), "https://www.facebook.com/gazillalounge");
        });

        clOpenInsta.setOnClickListener(v -> {
            openLink(getActivity(), "https://www.instagram.com/gazilla_gz");
        });

        clOpenMap.setOnClickListener(v -> {
            String adr = "Нижний Сусальный переулок 5с1";
            String uri1 = "geo:0,0?q=" + adr  ;
            Intent mapIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(uri1));
            startActivity(mapIntent);
        });

        clOpenWWW.setOnClickListener(v -> {
            openLink(getActivity(), "https://gazilla-lounge.ru");
        });

        clCallMe.setOnClickListener(v -> {
            Intent intent = new Intent(Intent.ACTION_DIAL);
            intent.setData(Uri.parse("tel:89652662222"));
            startActivity(intent);
        });

        return view;

    }

    public static void openLink(Activity activity, String url) {
        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
        List<ResolveInfo> resInfo = activity.getPackageManager().queryIntentActivities(intent, 0);

        if (resInfo.isEmpty()) return;

        for (ResolveInfo info: resInfo) {
            if (info.activityInfo == null) continue;
            if (VK_APP_PACKAGE_ID.equals(info.activityInfo.packageName)
                    || FACEBOOK_APP_PACKAGE_ID.equals(info.activityInfo.packageName)
                    ) {
                intent.setPackage(info.activityInfo.packageName);
                break;
            }
        }
        activity.startActivity(intent);
    }


}

File: app/src/main/java/com/gazilla/mihail/gazillaj/kotlin/activites/ContactsFragment.kt
package com.gazilla.mihail.gazillaj.kotlin.activites

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.arellomobile.mvp.MvpAppCompatFragment
import com.gazilla.mihail.gazillaj.R
import kotlinx.android.synthetic.main.fragment_contacts.*

class ContactsFragment: MvpAppCompatFragment() {

    private val VK_APP_PACKAGE_ID = "com.vkontakte.android"
    private val FACEBOOK_APP_PACKAGE_ID = "com.facebook.katana"


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_contacts, null)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        clOpenVK.setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://vk.com/gazilla_gz"))
            startActivity(intent)
        }

        clOpenFB.setOnClickListener {
            openLink(activity!!, "https://www.facebook.com/gazillalounge")
        }

        clOpenInsta.setOnClickListener { openLink(activity!!, "https://www.instagram.com/gazilla_gz") }

        clOpenMap.setOnClickListener {
            val adr = "Нижний Сусальный переулок 5с1"
            val uri1 = "geo:0,0?q=$adr"
            val mapIntent = Intent(Intent.ACTION_VIEW, Uri.parse(uri1))
            startActivity(mapIntent)
        }

        clOpenWWW.setOnClickListener {
            openLink(activity!!, "https://gazilla-lounge.ru")
        }

        clCallMe.setOnClickListener {
            val intent = Intent(Intent.ACTION_DIAL)
            intent.data = Uri.parse("tel:89652662222")
            startActivity(intent)
        }
    }

    private fun openLink(activity: Activity, url: String) {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
        val resInfo = activity.packageManager.queryIntentActivities(intent, 0)

        if (resInfo.isEmpty()) return

        for (info in resInfo) {
            if (info.activityInfo == null) continue
            if (VK_APP_PACKAGE_ID == info.activityInfo.packageName || FACEBOOK_APP_PACKAGE_ID == info.activityInfo.packageName) {
                intent.setPackage(info.activityInfo.packageName)
                break
            }
        }
        activity.startActivity(intent)
    }

}